### DISCUSSION TITLE
Datraloader Structure

### DISCUSSION THREAD

**Discussion by chwunderpers** [mrdbourke/pytorch-deep-learning#727] [Q&A]
Hi Daniel,
Hi everyone,

a quick question in section / chapter 03 - computer vision where you introduced the dataloader first It is clear to me, what this class is doing, nevertheless I do have questions when it comes to iterating through the dataloader.

In the training / test loop for the FahsionMNIST dataset, you are extracting information from the dataloader with this code.

```
for batch, (X_train,y_train) in enumerate(train_dataloader):
```

my question is now, from where do you know whats inside the dataloader? Couldn´t it be, that there is a z_train value in it as well? How do you know which order e.g batch, x, y you have to get the information? I struggle with this line as it seems to be magic that you know whats inside. I was not able to look up the structure of the dataloader. And why do you need a bracket for X and y?

I would appreciate if you can help me out here.

Best,
Chris

**Comment by shisirkha** [comment]
I could help. your The DataLoader function in torch.utils.data returns the shape of (BATCH_SIZE,actual_X,label)

in here BATCH_SIZE is = the photo amount per look by cpu or cuda
in here actual_X is = the train or test X
in here label is = class name means 0,1,9,3 etc.

#LovefromBangladeshAndShisir
